matrix/lambdas/api/core.py

```diff
@@ -55,11 +59,31 @@ def post_matrix(body: dict):
             }, requests.codes.accepted
 
 
+# TODO: write tests
 def get_matrix(request_id: str):
+    dynamo_handler = DynamoHandler()
+    # TODO: error handling: invalid request_id
+    ready, ready = dynamo_handler.increment_table_field(DynamoTable.STATE_TABLE,
+                                                        request_id,
+                                                        StateTableField.COMPLETED_REDUCER_EXECUTIONS.value,
+                                                        0)
+    if ready:
+        # TODO: error handling: missing zarr (corrupted zarr?)
+        s3_key = f"s3://{os.environ['S3_RESULTS_BUCKET']}/{request_id}.zarr"
+        return {'request_id': request_id,
+                'status': "Complete",
+                'key': s3_key,
+                'eta': "N/A",
+                'message': f"Request {request_id} has successfully completed. The resultant expression matrix can be "
+                           f"downloaded at the S3 location {s3_key}",
+                'links': [],
+                }, requests.codes.ok
+
+    # TODO: return appropriate code if result is not ready
     return {'request_id': request_id,
-            'status': "In Progress",
-            'key': "sample key",
-            'eta': "sample eta",
-            'message': "sample message",
+            'status': "In progress",
+            'key': "N/A",
+            'eta': "N/A",
+            'message': "Request {request_id} is still processing. Please try again later.",
             'links': [],
             }, requests.codes.ok
```

Comment:
is there a better http status code we can use here?

Reply:
Possibly 303 See Other?
https://github.com/requests/requests/blob/master/requests/status_codes.py

Reply:
**Pick your adventure** There is no standard.

Even Microsoft suggests both of the common patterns:

[Azure REST API](https://docs.microsoft.com/en-us/azure/architecture/best-practices/api-design#asynchronous-operations) recommends:
* a 202 response to the POST with the monitor in the Location: header
* a 303 response to the GET when the operation is successful and the created resource URI in the Location:header

[Microsoft REST API Guidelines](https://github.com/Microsoft/api-guidelines/blob/vNext/Guidelines.md#1327-the-typical-flow-polling) recommends:
* a 202 response to the POST with the monitor in a proprietary header
* a 200 response to the GET when the operation is successful with the created resource URI in the message body (_resourceLocation_)

Reply:
@mckinsel - If we continue to embed the created Resource URI in the message body, could we use a _self-documenting_ name (based on Dave Roger's questions). If _key_ was renamed _expression-matrix_ (or _matrix-location_) wouldn't that be relatively clear?